Backend/controller/NewWorldScraper/NewWorldScraper.js

```diff
@@ -60,102 +58,64 @@ async function fetchAllItemsOneCategory(category){
-    }
   };
 
-  try {
-    let response = await axios.request(allItemsConfig);
-    console.log("this request Passed");
-    console.log(url);
-    let data = response.data;
-    return data;
-  } catch (error){
-    console.error(`ERROR: ${error}`);
-    console.log("This request failed");
-    console.log(url);
-    throw error;
-  }
+  let response = await axios.request(allItemsConfig);
+  let data = response.data;
+  return data;
 }
 
-let failCounter = 0;
+/**
+ * @param {Array<String>} categories 
+ */
 async function fetchAllitems(categories){
-  const maxRetries = 5;
-  let retryDelay = 2000;
-
-  for (let retryCount = 0; retryCount < maxRetries; retryCount++){
-    try{
-      // we must wait untill all the prmises from the fetch of each category is resolved
-      await Promise.all(
-        //map the fetchAllItems function to each category
-        categories.map(async (category) => {
-          let itemsFromCategory = await fetchAllItemsOneCategory(category);
-          //save the fetched items into the products array
-          getAllItems(itemsFromCategory);
-        })
-      );
-      break;
-    }catch (error){
-      console.error(`Error occurred: ${error}`);
-      if (retryCount === maxRetries - 1) {
-        console.error('Max retries exceeded. Unable to fetch all items.');
-        break;
+  const MAX_RETRIES = 10;
+  const REQUEST_DELAY_SEP = 150;
+
+  let res = await Promise.all(categories.map(async (cat, index) => {
+    // Setup a delay between initial requests
+    await new Promise(resolve => setTimeout(resolve, index * REQUEST_DELAY_SEP));
+
+    for(let i = 0; i < MAX_RETRIES; i++) {
+      try {
+      console.log(`Sending request for items in category ${cat}`)
+        return await fetchAllItemsOneCategory(cat);
+      } catch {
+        console.warn(`Failed to fetch items for category ${cat}, retrying (attempt ${i + 2}/${MAX_RETRIES})`)
       }
-       // Exponential backoff - increase the delay before retrying
-       await new Promise((resolve) => setTimeout(resolve, retryDelay));
-
-       // Increase the delay for the next retry
-      retryDelay *= 2;
-
-
     }
-  }
+  }));
 
+  return res.flatMap(res => res.data.products.map(normaliseItem));
 }
 
 
-function getAllItems(allItemData){
-  try{
-    allItemData.data.products.forEach(function(oneProduct){
-      let productToPush = {};
-      id++;
-      productToPush.id = id;
-      productToPush.productID = oneProduct.productId;
-      productToPush.brand = oneProduct.brand;
-      productToPush.name = oneProduct.name;
-      productToPush.price = oneProduct.price;
-      productToPush.NonloyaltyCardPrice = oneProduct.nonLoyaltyCardPrice;
-      productToPush.quanityType = oneProduct.displayName;
-  
-      productsArray.push(productToPush)
+function normaliseItem(item){
```

Comment:
Just changed this function so that it now gets the relevant data, and doesn't use a global variable (apart from id)